metrics_api/lib/opentelemetry/metrics/instrument/synchronous_instrument.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Metrics
+    module Instrument
+      # https://opentelemetry.io/docs/reference/specification/metrics/api/#synchronous-instrument-api
+      class SynchronousInstrument
+        attr_reader :name, :unit, :description
+
+        # @api private
+        def initialize(name, unit: nil, description: nil)
```

Comment:
> - advice for implementations.
>
>Users can provide advice, but its up to their discretion. Therefore, this API needs to be structured to accept advice, but MUST NOT obligate the user to provide it.
>
>advice needs to be structured as described in [instrument advice](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advice), with parameters that are general and specific to a particular instrument kind. The API SHOULD NOT validate advice.

Did you want to work this in as a follow up PR? https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#synchronous-instrument-api

Reply:
Updated in https://github.com/open-telemetry/opentelemetry-ruby/pull/1442/commits/7cce415076ab44665b2ead3e4836de42796a21f5

I'm not sure how `advice` will be exactly used, but I think of it as an "options" Hash with recommended values. For example in docs, [the recommended bucket boundaries for a histogram](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advice). I imagine they will add standard names to be used in `advice`